packages/react/test/createRenderer.ts

```diff
@@ -23,13 +30,41 @@ export function createRenderer(globalOptions?: CreateRendererOptions): BaseUITes
   const createRendererResult = sharedCreateRenderer(globalOptions);
   const { render: originalRender } = createRendererResult;
 
-  const render = async (element: React.ReactElement<DataAttributes>, options?: RenderOptions) =>
-    act(async () => {
+  const render = async (element: React.ReactElement<DataAttributes>, options?: RenderOptions) => {
+    const renderResult = await act(async () => {
       const result = await originalRender(element, options);
       await flushMicrotasks();
       return result;
     });
 
+    if (!isJSDOM) {
+      const { userEvent: vitestUserEvent } = await import('@vitest/browser/context');
+
+      const originalUser = vitestUserEvent.setup();
+
+      const proxyHandler: ProxyHandler<any> = {
+        get: (target, prop) => {
+          if (typeof target[prop] !== 'function') {
+            return target[prop];
+          }
+
+          return async (...args: unknown[]) => {
+            await act(async () => target[prop](...args));
```

Comment:
This isn't ideal. Perhaps we could change the implementation of `act` to no-op when running in the browser and get rid of this.